front/components/vaults/VaultSideBarMenu.tsx

```diff
@@ -431,33 +434,87 @@ const VaultDataSourceViewItem = ({
   item,
   owner,
   vault,
+  node,
 }: {
   item: DataSourceViewType;
   owner: LightWorkspaceType;
   vault: VaultType;
+  node?: DataSourceViewContentNode;
 }): ReactElement => {
   const router = useRouter();
+  const [isExpanded, setIsExpanded] = useState(false);
 
-  const LogoComponent = getConnectorProviderLogoWithFallback(
-    item.dataSource.connectorProvider,
-    FolderIcon
-  );
-  const dataSourceViewPath = `/w/${owner.sId}/vaults/${vault.sId}/categories/${item.category}/data_source_views/${item.sId}`;
+  const { isNodesLoading, nodes } = useDataSourceViewContentNodes({
+    dataSourceView: item,
+    owner,
+    parentId: node ? node?.internalId : undefined,
+    viewType: "documents",
+  });
+
+  // Load the currently selected node from router.query.parentId
+  const {
+    nodes: [selected],
+  } = useDataSourceViewContentNodes({
+    dataSourceView: router.query.parentId ? item : undefined,
+    owner,
+    internalIds: [router.query.parentId as string],
+    viewType: "documents",
+  });
+
+  const basePath = `/w/${owner.sId}/vaults/${vault.sId}/categories/${item.category}/data_source_views/${item.sId}`;
+
+  // isAncestorToCurrentPage is true if :
+  // 1. The current path matches the basePath
+  // 2. Either the current node is the root (!node) or the current node is a parent of the selected node
+  const isAncestorToCurrentPage =
+    router.asPath.startsWith(basePath) &&
+    (!node || (node && selected?.parentInternalIds?.includes(node.internalId)));
+
+  // Unfold the folder if it's an ancestor of the current page.
+  useEffect(() => {
+    if (isAncestorToCurrentPage) {
```

Comment:
I'll let you check with @Duncid but IIRC we said that browsing files and folders from the right side should not expand items on the left side.

Reply:
![Sep-19-2024 13-28-28](https://github.com/user-attachments/assets/43651618-912d-46cd-af11-3ba91944073f)

Reply:
According to @Duncid we have the expected behaviour here